lib/containers/container_images.pm

```diff
@@ -185,6 +186,32 @@ sub test_opensuse_based_image {
     }
 }
 
+sub verify_userid_on_container {
+    my ($runtime, $image) = @_;
+    record_info "host uid", script_output "echo \$UID";
+
+    record_info "root user inside", "process runs in the user's host uid";
+    my $cid = script_output "$runtime run -d --rm --name test1 $image sleep infinity";
+    assert_script_run "$runtime exec -it $cid id";
+    assert_script_run "$runtime exec -it $cid cat /proc/self/uid_map";
+    validate_script_output "$runtime top $cid user huser", sub { /root\s+1000/ };
+    validate_script_output "$runtime top $cid capeff",     sub { /setuid/i };
+
+    record_info "regular user inside", "process runs in the range of subuids";
+    $cid = script_output "$runtime run -d --rm --name test2 --user 1000 $image sleep infinity";
+    assert_script_run "$runtime exec -it $cid id";
+    assert_script_run "$runtime exec -it $cid cat /proc/self/uid_map";
+    validate_script_output "$runtime top $cid user huser", sub { /1000\s+200999/ };
+    validate_script_output "$runtime top $cid capeff",     sub { /none/ };
+
+    record_info "root inside", "process runs with the same uid as host user";
```

Comment:
What is the difference between this `root inside` and the `root user inside` above?

Reply:
`root inside`: regular user run with the default container user
`root user inside`: with the default container user again but process runs with the same uid as host user

Reply:
> `root inside`: regular user run with the default container user

so maybe "default container user" will be more clear  ?

> `root user inside`: with the default container user again but process runs with the same uid as host user

correct me if I wrong but processes inside a container by cgroups so it does not matter if it is the same id or different one , so I am not sure what we try to test here ? @Vogtinator  do you think such differentiation run something as default container user and that run the same as uid of host user make any sense ?

Reply:
sorry I skipped the verb :)  "processes inside a container **are isolated** by cgroups "

Reply:
This definitely needs some comments explaining each scenario.

Reply:
please i am open for suggestion if the descriptions are not explanatory enough

Reply:
i changed the record_info. please check changes.